### ISSUE TITLE
Outlet state does not update when last command is a voice command 

### ISSUE THREAD

**Issue by Chrischi-** [mikejac/node-red-contrib-google-smarthome#77]
The Outlet node doesn't know his actual state when state set trough voice and changed via (input) Node-Red payload.

Initial state **Off**
"Hey Google, turn "Outlet-Name" on" 

switches state to **On**

Set Payload via Node-Red Flow to Off

```
    msg.topic = 'set'
    msg.payload = {
      on: false
    }
```
Outlet-Node ignores payload and does nothing. Even the icon remains unchanged. Using voice works always.

If I manually set an On payload (the same state as published via voice) again and after that the Off payload, 
then the node notices the difference and is working.

It seems that the IN state of the node is not overwritten and therefore discards the following payload because
he thinks the node already has this state.

The whole thing does not work the other way around either.

I have tested everything with the similar Light Node (On/Off) and this works as expected.

**Comment by Caprico85** [comment]
Fix this (and similar problems on other devices) in v0.0.52.

**Comment by Chrischi-** [comment]
thanks @Caprico85 but this does not solve the problem that the node states are inconsistent between 
"respond to inputs from NodeRED" and "called when state is updated from Google Assistant".

I think "called when state is updated from Google Assistant" should contain the 
following code (e.g. outlet.js; between 103 and 105):
```
            node.states.on = states.on;
            node.states.online = states.online;
```
So `node.states.on` and `node.states.online` are synchronous between the two control options and the if-statement at line 185 `if (node.states.on !== on || node.states.online !== online)` will work as intended.

I think this has to be changed for all devices to work properly.

What do you think?

**Comment by Caprico85** [comment]
You are right. In 0.0.52 I only fixed the status icon handling (and somehow not even that seems to work correctly).
The `updated()` functions on all devices needs to be changed to set node.states.

**Comment by Chrischi-** [comment]
I am almost sure that you have only "moved" the problem.  
As soon as the `node.states` are synchronised across the functions, the problem with the "icons" will also be solved. 

Maybe we can rename your `updateStatusIcon` function and use it for the `node.states` updates as well?

**Comment by Chrischi-** [comment]
Fixed State with #87

**Comment by Apple-G** [comment]
I have the same Issue with the Light-Node (On/Off-Light) now, 
the state cannot be changed when the state changed through voice
